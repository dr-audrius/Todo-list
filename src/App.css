@import url("https://fonts.googleapis.com/css?family=Monoton");
@import url("https://fonts.googleapis.com/css?family=Cabin|Pacifico");
* {
  margin: 0;
  padding: 0 "";
}

body {
  background: #FFFFFF;
}

.wrap {
  background: #bbd1c0;
  width: 700px;
  margin: 0 auto;
  overflow: auto;
  position: relative;
}

h2 {
  color: #3e496b;
  font-size: 20px;
  font-family: "Arial", cursive;
}

h2,
form {
  background: #6c8571;
  margin: auto;
  padding: 10px 0;
  text-align: center;
}

form {
  padding: 0;
}

input {
  border: none;
  padding-top: 15px;
  padding-bottom: 15px;
  width: 100%;
}

input:focus {
  outline: none;
}

input[type=button] {
  margin-left: 15px;
  width: 130px;
  height: 30px;
  padding: 0px;
}

ul.items {
  width: 90%;
  margin: 50px auto;
  padding: 0;
  float: center;
}
ul.items li {
  list-style: none;
  border-bottom: 1px solid #000000;
  padding: 10px;
  font-size: 20px;
  font-family: "Arial", sans-serif;
  position: relative;
}
ul.items li span {
  font-size: 13px;
  margin-left: 5px;
  position: relative;
  color: #000000;
}
ul.items li span:after {
  position: absolute;
  content: "";
  display: block;
  width: 13px;
  height: 13px;
  top: 0;
  left: -5px;
  border: 2px solid #000000;
  border-radius: 50%;
}

/*a {
  text-decoration: none;
  position: absolute;
  background: #ecd641;
  color: #3e496b;
  font-weight: bolder;
  right: 0;
  padding: 3px;
}*/
.button {
  float: right;
}

.menu {
  width: 100%;
  padding: 0;
  overflow: auto;
}
.menu li {
  list-style: none;
}
.menu a {
  text-decoration: none;
  text-align: center;
  padding: 10px;
  display: inline-block;
  width: 27%;
  float: left;
  color: black;
  font-weight: bolder;
  border-bottom: 2px solid #bbd1c0;
}
.menu a:hover {
  background: #6c8571;
  color: white;
}

/*# sourceMappingURL=App.css.map */
