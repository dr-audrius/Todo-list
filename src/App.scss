@import url("https://fonts.googleapis.com/css?family=Monoton");
@import url("https://fonts.googleapis.com/css?family=Cabin|Pacifico");
* {
  margin: 0;
  padding: 0 "";
}
$white: #FFFFFF;
$black: #000000;
$dark-blue: #3e496b;
$yellow: #ecd641;
$red: #ea315b;
$light-blue:  #44b1d9;
$dark-green:  #6c8571;
$light-green:  #bbd1c0;


body {
  background: $white;
}
.wrap {
  background: $light-green;
  width: 700px;
  margin: 0 auto;
  overflow: auto;
  position: relative;
}
h2 {
  color: $dark-blue;
  font-size: 20px;
  font-family: "Arial", cursive;
}
h2,
form {
  background: $dark-green;
  margin: auto;
  padding: 10px 0;
  text-align: center;
}
form {
  padding: 0;
}
input {
  border: none;
  padding-top: 15px;
  padding-bottom: 15px;
  width: 100%;
}
input:focus {
  outline: none;
}
input[type=button]{
  margin-left: 15px;
  width: 130px;
  height: 30px;
  padding: 0px;
}
ul.items {
  width: 90%;
  margin: 50px auto;
  padding: 0;
  float:center;

  li {
    list-style: none;
    border-bottom: 1px solid $black;
    padding: 10px;
    font-size: 20px;
    font-family: "Arial", sans-serif;
    position: relative;
  }
  li span {
    font-size: 13px;
    margin-left: 5px;
    position: relative;
    color: $black;
  }
  li span:after {
    position: absolute;
    content: "";
    display: block;
    width: 13px;
    height: 13px;
    top: 0;
    left: -5px;
    border: 2px solid $black;
    border-radius: 50%;
  }
}
/*a {
  text-decoration: none;
  position: absolute;
  background: #ecd641;
  color: #3e496b;
  font-weight: bolder;
  right: 0;
  padding: 3px;
}*/

.button{
  float: right;
}

.menu{
  width:100%;
  padding:0;
  overflow: auto;
  li{
    list-style:none;
  }
  a{
    text-decoration: none;
    text-align: center;
    padding: 10px;
    display: inline-block;
    width:27%;
    float:left;
    color: black;
    font-weight: bolder;
    border-bottom: 2px solid $light-green;
    &:hover{
      background: $dark-green;
      color:white;
    }
  }
}