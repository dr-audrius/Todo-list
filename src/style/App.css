@import url("https://fonts.googleapis.com/css?family=Monoton");
@import url("https://fonts.googleapis.com/css?family=Cabin|Pacifico");
* {
  margin: 0;
  padding: 0 "";
}

body {
  background: #FFFFFF;
}

.wrap {
  background: #bbd1c0;
  width: 700px;
  margin: 0 auto;
  overflow: auto;
  position: relative;
}

h2 {
  color: #3e496b;
  font-size: 20px;
  font-family: "Arial", cursive;
}

h2,
form {
  background: #6c8571;
  margin: auto;
  padding: 10px 0;
  text-align: center;
}

input {
  border: none;
  padding-top: 15px;
  padding-bottom: 15px;
  width: 100%;
}

input:focus {
  outline: none;
}

input[type=button] {
  margin-left: 15px;
  width: 130px;
  height: 30px;
  padding: 0px;
}

ul.items {
  width: 90%;
  margin: 50px auto;
  padding: 0;
  float: center;
}
ul.items li {
  list-style: none;
  border-bottom: 1px solid #000000;
  padding: 10px;
  font-size: 20px;
  font-family: "Arial", sans-serif;
  position: relative;
}
ul.items li span {
  font-size: 13px;
  margin-left: 5px;
  position: relative;
  color: #000000;
}
ul.items li span:after {
  position: absolute;
  content: "";
  display: block;
  width: 13px;
  height: 13px;
  top: 0;
  left: -5px;
  border: 2px solid #000000;
  border-radius: 50%;
}

/*a {
  text-decoration: none;
  position: absolute;
  background: #ecd641;
  color: #3e496b;
  font-weight: bolder;
  right: 0;
  padding: 3px;
}*/
.menu {
  width: 100%;
  padding: 0;
  overflow: auto;
}
.menu li {
  list-style: none;
}
.menu a {
  text-decoration: none;
  text-align: center;
  padding: 10px;
  display: inline-block;
  width: 27%;
  float: left;
  color: black;
  font-weight: bolder;
  border-bottom: 2px solid #bbd1c0;
}
.menu a:hover {
  background: #6c8571;
  color: white;
}

.auth {
  margin: auto;
}
.auth .facebook {
  background: #4B64B1;
  color: white;
  width: 90%;
  margin: 30px auto auto auto;
  text-align: center;
  padding: 20px 0;
}
.auth form {
  margin-top: 30px;
  text-align: left;
  width: 90%;
}
.auth, .auth form {
  background: #6c8571;
}
.auth input {
  width: 100%;
  margin-top: 5px;
}
.auth input[type=submit] {
  margin: 0;
  background: #cccccc;
  padding: 20px 0;
  color: black;
  margin-top: 20px;
  text-align: center;
}
.auth input[type=register] {
  margin: 0;
  background: #cccccc;
  padding: 20px 0;
  color: black;
  margin-top: 20px;
  text-align: center;
}
.auth label {
  display: block;
  margin-top: 10px;
}

.logOut {
  text-decoration: none;
  position: absolute;
  background: #bbd1c0;
  color: #3E496B;
  font-weight: bolder;
  right: 0;
  top: 0;
  padding: 3px;
}

/*# sourceMappingURL=App.css.map */
