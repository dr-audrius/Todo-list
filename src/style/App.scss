@import url("https://fonts.googleapis.com/css?family=Monoton");
@import url("https://fonts.googleapis.com/css?family=Cabin|Pacifico");
* {
  margin: 0;
  padding: 0 "";
}
$white: #FFFFFF;
$black: #000000;
$dark-blue: #3e496b;
$yellow: #ecd641;
$red: #ea315b;
$light-blue:  #44b1d9;
$dark-green:  #6c8571;
$light-green:  #bbd1c0;


body {
  background: $white;
}
.wrap {
  background: $light-green;
  width: 700px;
  margin: 0 auto;
  overflow: auto;
  position: relative;
}
h2 {
  color: $dark-blue;
  font-size: 20px;
  font-family: "Arial", cursive;
}
h2,
form {
  background: $dark-green;
  margin: auto;
  padding: 10px 0;
  text-align: center;
}
input {
  border: none;
  padding-top: 15px;
  padding-bottom: 15px;
  width: 100%;
}
input:focus {
  outline: none;
}
input[type=button]{
  margin-left: 15px;
  width: 130px;
  height: 30px;
  padding: 0px;
}
ul.items {
  width: 90%;
  margin: 50px auto;
  padding: 0;
  float:center;

  li {
    list-style: none;
    border-bottom: 1px solid $black;
    padding: 10px;
    font-size: 20px;
    font-family: "Arial", sans-serif;
    position: relative;
  }
  li span {
    font-size: 13px;
    margin-left: 5px;
    position: relative;
    color: $black;
  }
  li span:after {
    position: absolute;
    content: "";
    display: block;
    width: 13px;
    height: 13px;
    top: 0;
    left: -5px;
    border: 2px solid $black;
    border-radius: 50%;
  }
}
/*a {
  text-decoration: none;
  position: absolute;
  background: #ecd641;
  color: #3e496b;
  font-weight: bolder;
  right: 0;
  padding: 3px;
}*/

.menu{
  width:100%;
  padding:0;
  overflow: auto;
  li{
    list-style:none;
  }
  a{
    text-decoration: none;
    text-align: center;
    padding: 10px;
    display: inline-block;
    width:27%;
    float:left;
    color: black;
    font-weight: bolder;
    border-bottom: 2px solid $light-green;
    &:hover{
      background: $dark-green;
      color:white;
    }
  }
}


.auth{
  margin: auto;
  .facebook {
    background: #4B64B1;
    color: white;
    width: 90%;
    margin: 30px auto auto auto;
    text-align: center;
    padding: 20px 0;
  }
  form {
    margin-top: 30px;
    text-align: left;
    width: 90%;
  }
  &, form {
    background: $dark-green;
  }
  input {
    width: 100%;
    margin-top: 5px;
  }
  input[type="submit"]{
    margin: 0;
    background: lighten(black, 80%);
    padding: 20px 0;
    color: black;
    margin-top: 20px;
    text-align: center;
  }
  input[type="register"]{
    margin: 0;
    background: lighten(black, 80%);
    padding: 20px 0;
    color: black;
    margin-top: 20px;
    text-align: center;
  }
  label {
    display: block;
    margin-top: 10px;
  }
}
.logOut {
  text-decoration: none;
  position: absolute;
  background: $light-green;
  color: #3E496B;
  font-weight: bolder;
  right: 0;
  top:0;
  padding: 3px;
}