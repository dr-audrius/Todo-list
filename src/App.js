import React, { Component } from 'react'
import { BrowserRouter, Route } from 'react-router-dom';
import ItemsComponent from './components/ItemsComponent';
import Menu from './components/Menu';
import './style/App.css'
import { fire } from './fire';
import { ToastContainer, toast } from 'react-toastify';
import 'react-toastify/dist/ReactToastify.css';

class App extends Component {
  state = {
    items: {},
    authenticated: false,
    loading: true
  }
  itemsRef = '';

  EmailAndPasswordAuthentication = (e) => {
    e.preventDefault()
    const email = this.emailInput.value;
    const password = this.passwordInput.value;
    fire.auth().fetchProvidersForEmail(email)
      .then(provider => {
        if (provider.length === 0) {
          toast.success("Please enter email and password"), { autoClose: 20000 }
        } else if (provider.indexOf("password") === -1) {
          toast.error("Pasword for email:  " + email + " is incorrect")
        } else {
          return fire.auth().signInWithEmailAndPassword(email, password)
        }
      })
      .catch((error) => {
        toast.error("Ooops:  " + error.message)
      })
  }

  EmailAndPasswordRegistration = (e) => {
    e.preventDefault()
    const email = this.emailInput.value;
    const password = this.passwordInput.value;
    fire.auth().fetchProvidersForEmail(email)
      .then((provider) => {
        if (provider.length === 0) {
          return fire.auth().createUserWithEmailAndPassword(email, password)
            .then(toast.success("Success. Account with email: " + email + " has been created"), { autoClose: 20000 })
        } else if (provider.indexOf("email") === -1) {
          toast.success("Account with email: " + email + " is already present. Please login instead of register."), { autoClose: 30000 }
        }
      })
      .catch((error) => {
        toast.error("Ooops:  " + error.message)
      })
  }

  componentDidMount() {
    this.removeAuthListener = fire.auth().onAuthStateChanged(user => {
      if (user) {
        this.itemsRef = fire.database().ref(`items/${user.uid}`)
        this.itemsRef.on('value', data => {
          this.setState({
            authenticated: true,
            items: data.val(),
            loading: false
          })
        })
      } else {
        this.setState({
          authenticated: false,
          loading: false
        })
      }
    })
  }
  logOut = () => {
    fire.auth().signOut().then((user) => {
      this.setState({ items: null })
    })
  }
  componentWillUnmount() {
    fire.removeBinding(this.itemsRef)
  }
  completeItem = (id) => {
    this.itemsRef.update({
      [id]: {
        ...this.state.items[id],
        completed: true
      }
    })
  }
  changeItem = (id) => {
    console.log("STATUS", this.state.items[id].completed)
    this.itemsRef.update({
      [id]: {
        ...this.state.items[id],
        completed: !this.state.items[id].completed
      }
    })
  }
  deleteItem = (id) => {
    this.itemsRef.update({
      [id]: null
    })
  }
  removeItem = (id) => {
    this.itemsRef.child(id).remove()
  }
  addItem = (e) => {
    e.preventDefault();
    this.itemsRef.push({
      item: this.todoItem.value,
      completed: false
    })
  }
  render() {
    if (this.state.loading) {
      return (<h3>Loading</h3>)
    }
    return (
      <div>
        <ToastContainer />
        <BrowserRouter>
          <div className="wrap">
            <h2>To do</h2>
            <Menu
              logOut={this.logOut}
              authenticated={this.state.authenticated}
              authWithFacebook={this.authWithFacebook}
              emailInput={el => this.emailInput = el}
              passwordInput={el => this.passwordInput = el}
              EmailAndPasswordAuthentication={this.EmailAndPasswordAuthentication}
              EmailAndPasswordRegistration={this.EmailAndPasswordRegistration}
            />
            <ul className="menu">
            </ul>
            <Route exact path="/"
              render={props =>
                <ItemsComponent
                  items={this.state.items}
                  all={false}
                  done={false}
                  action={this.completeItem}
                  addItem={this.addItem}
                  changeItem={this.changeItem}
                  removeItem={this.removeItem}
                  inputRef={el => this.todoItem = el}
                  authenticated={this.state.authenticated}
                />
              } />
            <Route exact path="/completed"
              render={props =>
                <ItemsComponent
                  items={this.state.items}
                  all={false}
                  done={true}
                  changeItem={this.changeItem}
                  removeItem={this.removeItem}
                  authenticated={this.state.authenticated}
                />
              } />
            <Route exact path="/all"
              render={props =>
                <ItemsComponent
                  items={this.state.items}
                  addItem={this.addItem}
                  changeItem={this.changeItem}
                  removeItem={this.removeItem}
                  all={true}
                  inputRef={el => this.todoItem = el}
                  authenticated={this.state.authenticated}
                />
              } />
          </div>
        </BrowserRouter>
      </div>
    );
  }
}
export default App;