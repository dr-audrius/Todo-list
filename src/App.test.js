import App from './App'

it('shallow matches snapshot', () => {
  const wrapper = shallow(
    <App />
  )
  expect(wrapper).toMatchSnapshot();
});


it('render matches snapshot', () => {
  const wrapper = render(
    <App />
  )
  expect(wrapper).toMatchSnapshot();
});