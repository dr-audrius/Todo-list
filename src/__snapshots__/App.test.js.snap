// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render matches snapshot 1`] = `
<h3>
  Loading
</h3>
`;

exports[`shallow matches snapshot 1`] = `
<h3>
  Loading
</h3>
`;
